File: Javascript/Algorithms/1704.js
let halvesAreAlike = s => {
    let vowels = new Set("aAoOiIuUeE");
    const length = s.length,
          mid = Math.floor(length / 2);
    let count = 0;
    
    for (let i = 0; i < mid; i++){
        if (vowels.has(s[i])) {
            count++;
        }
    }
    for (let i = mid; i < length; i++){
        if (vowels.has(s[i])) {
            count--;
            if (count < 0){
                return false;
            }
        }
    }
    return count == 0;
};

File: C/Algorithms/1704.c
#include <stdbool.h>
#include <string.h>

bool halvesAreAlike(char *s) {
  int len = strlen(s) - 1;
  int mid = (len / 2) + 1, count = 0;
  char *vowels = "aAoOuUiIeE", *runner;
  // Go through first part.
  while (mid--) {
    runner = vowels;
    while (*runner) {
      if (*s == *runner++) {
        count++;
        break;
      }
    }
    s++;
  }
  // Continue until the end.
  while (*s) {
    runner = vowels;
    while (*runner) {
      if (*s == *runner++) {
        count--;
        if (count < 0)
          return 0;
        break;
      }
    }
    s++;
  }
  return count == 0;
}
